modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java

```diff
@@ -134,6 +142,18 @@ protected void validateOperationConfiguration(ConfigurationProvider configuratio
     }
   }
 
+  private void validatePolicyIsolation(ConfigurationProvider configurationProvider) {
+    if (featureFlaggingService.isEnabled(ENABLE_POLICY_ISOLATION)
+        && muleContext.getArtifactType().equals(POLICY)
+        && configurationProvider.getExtensionModel() != extensionModel) {
+      throw new IllegalOperationException(format(
```

Comment:
add a TODO to make this validation at the AST level, before actually initializing the app: https://www.mulesoft.org/jira/browse/MULE-19350